### ISSUE TITLE
Compile to take advantage of the new 396 megahertz processor.

### ISSUE THREAD

**Issue by Destroyinator69420** [Vogtinator/crafti#14]
I have been playing Crafti seriously on and off for about a month, and I have noticed that it has not been compiled for the new Processor in the TI Nspire CX II. It would run much faster if it were compiled to so do. If that is possible, it would also be possible to add working tools, pistons, observers, dispensers, a bow and arrow, and a peaceful survival mode, right?

**Comment by Vogtinator** [comment]
The processor core is the same (arm926ej-s), just with a higher frequency. That means recompiling wouldn't change anything.

I'm planning to release a new version soon though, which is built with a newer compiler and contains some more optimizations.

> If that is possible, it would also be possible to add working tools, pistons, observers, dispensers, a bow and arrow, and a peaceful survival mode, right?

Technically yes, but quite a lot of work. Tools, dispensers, bow arrows and survival mode would require an implementation of items and crafting and some mobs as well.

**Comment by Tristan1972** [comment]
Can we add wool and stairs? And yes, I like those options you said are possible. Can we also add a vanilla low-resolution texture pack from Minecraft? Can you add an fps counter like fraps? Maybe make the options look more like Minecraft? 

If we were able to add mobs, Could you make commands and day/night cycle? (I think this all would take you a year or two to get it all to work). If you were adding stuff from Minecraft, What version would you port it from?

Can you add a world border so if someone goes too far out, The ram does not max out?

Can you add quartz? falling sand and gravel? Like things from Minecraft itself?

**Comment by Vogtinator** [comment]
> Can we also add a vanilla low-resolution texture pack from Minecraft?

You can already do that! Just save the vanilla `terrain.png` in PPM format as `ndless/crafti.ppm.tns`.

> Can you add an fps counter like fraps?

That's in the next version already.

> Can you add a world border so if someone goes too far out, The ram does not max out?

That shouldn't happen in practice, because that far away the graphics get glitchy before it's even close to run out of RAM.

> If you were adding stuff from Minecraft, What version would you port it from?

No version in particular, everything needs to be done from scratch.

**Comment by Tristan1972** [comment]
How do I change this? I am not a developer, So I do not know what I am doing.

**Comment by Bluebotlabz** [comment]
> How do I change this? I am not a developer, So I do not know what I am doing.

If you are still interested in this, there is some work-in-progress documentation on my fork [Bluebotlaboratories/crafti](https://github.com/Bluebotlaboratories/crafti/wiki)

The documentation is still under development and is unfinished, however, all pages with working hyperlinks on the sidebar are complete.

Also note, that the documentation is specific to my fork which has some modified settings such as makefile parameters and also some renamed renderer-related variable names

**Comment by maj113** [comment]
> If you're still interested in this, my fork has an ALPHA release with basic pistons, you can build it yourself for more recent features but I will be making another preview release soon with fully working pistons and improved wool (changes are in repo but binaries are not yet available)

Since the Artifacts expired do you by any chance have a copy somewhere?

**Comment by Bluebotlabz** [comment]
yes, check the latest alpha release: https://github.com/Bluebotlaboratories/crafti/releases/tag/v1.3-CR13PR2
it has all the aforementioned features

**Comment by Bluebotlabz** [comment]
any commits since that release are unstable

**Comment by maj113** [comment]
> yes, check the latest alpha release: https://github.com/Bluebotlaboratories/crafti/releases/tag/v1.3-CR13PR2 it has all the aforementioned features

Ah, stupid me only checked the release ones and not prerelease ;P
